client/src/NetworkClient.ts

```diff
@@ -1,39 +1,52 @@
 import { EventClient, EventCallback } from '@nimiq/rpc-events';
 
-export type PlainTransaction = {
-    sender: string,
-    senderPubKey: Uint8Array,
-    recipient: string,
-    value: number,
-    fee: number,
-    validityStartHeight: number,
-    signature: Uint8Array,
-    extraData?: string | Uint8Array,
+export interface PlainTransaction {
+    sender: string;
+    senderPubKey: Uint8Array;
+    recipient: string;
+    value: number; // in NIM
+    fee: number; // IN NIM
+    validityStartHeight: number;
+    signature: Uint8Array;
+    extraData?: string | Uint8Array;
 }
 
-export type DetailedPlainTransaction = {
-    sender: string,
-    recipient: string,
-    value: number,
-    fee: number,
-    extraData: Uint8Array,
-    hash: string,
-    blockHeight: number,
-    blockHash: string,
-    timestamp: number,
-    validityStartHeight: number,
+export interface DetailedPlainTransaction {
+    sender: string;
+    recipient: string;
+    value: number; // in NIM
+    fee: number; // IN NIM
+    extraData: Uint8Array;
+    hash: string; // base64
+    blockHeight: number;
+    blockHash?: string; // base64
+    timestamp: number;
+    validityStartHeight: number;
 }
 
-export type PlainVestingContract = {
-    address: string,
-    owner: string,
-    start: number,
-    stepAmount: number,
-    stepBlocks: number,
-    totalAmount: number,
+export interface PlainVestingContract {
+    address: string;
+    owner: string;
+    start: number;
+    stepAmount: number;
+    stepBlocks: number;
+    totalAmount: number;
 }
 
-export class NetworkClient {
+class NetworkClient {
+    public static get Instance(): NetworkClient {
+        if (!NetworkClient._instance) {
+            NetworkClient._instance = new NetworkClient();
+        }
+        return NetworkClient._instance;
+    }
+
+    public static hasDefaultInstance() {
```

Comment:
What do you think about naming this `hasInstance` instead?

Reply:
Ok, I renamed it to `hasInstance` but therefore also changed the semantics to really only allow one instance.